### PR TITLE
BB-2

### PR DESCRIPTION
**Pull request by hladkky**

### CODE REVIEW

File: pages/_app.tsx [hunk 1]
```diff
@@ -1,11 +1,24 @@
 // import App from "next/app";
```
Comment (Nivolves) [inline pages/_app.tsx]: remove comments

File: pages/_app.tsx [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
 import Head from 'next/head';
 import { AppProps } from 'next/app';
+import '../styles/index.scss';
 
 const MyApp: React.FC<AppProps> = ({ Component, pageProps }): JSX.Element => {
   return (
     <>
       <Head>
+        {/* <link
```
Comment (Nivolves) [inline pages/_app.tsx]: remove comments

File: src/components/header/desktopHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  const menuRef = useRef<HTMLDivElement>(null);
+  const searchRef = useRef<HTMLDivElement>(null);
+
+  const changeCurMenu = (id: number) => {
+    if (!desktopMenuOpen) setDesktopMenuOpen(true);
+    else if (curMenu === id) setDesktopMenuOpen(false);
+    setCurMenu(id);
+  };
+
+  const handleClickOutside = useCallback(
```
Comment (Nivolves) [inline src/components/header/desktopHeader.tsx]: why is useCallback there?

File: src/components/header/mobileMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+          role="presentation"
+          onClick={close}
+        >
+          <CrossSVG />
+        </div>
+        <SearchInput />
+        <p role="presentation" onClick={() => setStoreListOpen(!storeListOpen)}>
+          Магазин
+        </p>
+        {storeListOpen ? (
```
Comment (Nivolves) [inline src/components/header/mobileMenu.tsx]: use storeListOpen && (...)

File: src/components/header/aboutList.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { AboutListData } from '../../constants/header';
```
Comment (Nivolves) [inline src/components/header/aboutList.tsx]: header folder with a capital letter

File: styles/layouts/header/_header.scss [hunk 6]
```diff
@@ -0,0 +1,89 @@
+@import './mobileMenu';
```
Comment (Nivolves) [inline styles/layouts/header/_header.scss]: why u store scss files which refer to header in global folder?

File: styles/variables/_variables.scss [hunk 7]
```diff
@@ -0,0 +1,9 @@
+$primary-font: Futura;
```
Comment (Nivolves) [inline styles/variables/_variables.scss]: css vars, not scss

File: styles/layouts/header/_header.scss (hunk 6 above)
Comment (Nivolves) [reply styles/layouts/header/_header.scss]: u can store css vars files, global styles and mixins in global style folder, etc but not files which refers to a component

File: src/components/header/desktopHeader.tsx (hunk 3 above)
Comment (hladkky) [reply src/components/header/desktopHeader.tsx]: Зробив як ти в РКЦ

File: styles/layouts/header/_header.scss (hunk 6 above)
Comment (hladkky) [reply styles/layouts/header/_header.scss]: Тобто усі файли scss переносити прямо в папку компонентів -> папку відповідного компетенту?

File: src/components/header/desktopHeader.tsx (hunk 3 above)
Comment (Nivolves) [reply src/components/header/desktopHeader.tsx]: ну, видимо я не правильно сделал там

File: styles/layouts/header/_header.scss (hunk 6 above)
Comment (Nivolves) [reply styles/layouts/header/_header.scss]: Как в РКЦ было 1 компонент <=> 1 scss файл

File: src/components/Header/AboutList.tsx [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import { AboutListData } from '../../constants/header';
```
Comment (Nivolves) [inline src/components/Header/AboutList.tsx]: AboutList and StoreList are the same. Add props to one of components and remove other.

File: src/components/Header/MobileMenu.tsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+          role="presentation"
+          onClick={close}
+        >
+          <CrossSVG />
+        </div>
+        <SearchInput />
+        <p role="presentation" onClick={() => setStoreListOpen(!storeListOpen)}>
+          Магазин
+        </p>
+        {storeListOpen && (
```
Comment (Nivolves) [inline src/components/Header/MobileMenu.tsx:30]: use css to hide a component(it better for SEO and u can add animation) and add animation when about and shop switch like on bevza.com

File: src/components/Header/LanguageSelector.tsx [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+import { LANGUAGES } from '../../constants/languages';
+
+const LanguageSelector: React.FC = () => {
+  const { language, setLanguage } = useContext(context) as IContext;
+
+  const languagesItems = Object.keys(LANGUAGES);
+
+  const [droplistOpen, setDroplistOpen] = useState<boolean>(false);
+
+  const handleLanguageChange = (newLang: string) => {
```
Comment (Nivolves) [inline src/components/Header/LanguageSelector.tsx:16]: and here u need to use useCallback hook

File: src/components/Header/Sublist.tsx [hunk 11]
```diff
@@ -0,0 +1,13 @@
+import { ISublistProps } from './Types';
```
Comment (Nivolves) [inline src/components/Header/Sublist.tsx:1]: remove it and AboutList

File: src/components/Header/DesktopHeader.tsx [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+import LanguageSelector from './LanguageSelector';
+import Logo from './Logo';
+import SearchInput from './SearchInput';
+import Sublist from './Sublist';
+
+import { AboutListData, StoreListData } from '../../constants/header';
+
+const DesktopHeader: React.FC = () => {
+  const [desktopMenuOpen, setDesktopMenuOpen] = useState<boolean>(false);
+  const [curMenu, setCurMenu] = useState<number>(1);
```
Comment (Nivolves) [inline src/components/Header/DesktopHeader.tsx]: u can implement it more clearly like this  const [curMenu, setCurMenu] = useState<"store" | "about">("store");

**Pull request merged**